src/routes/Profile/get-referrer-link.ts

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+
+export const getReferrerLink = async (account: string) => {
+  const baseUrl = process.env.REACT_APP_REFERRAL_AND_SIGNATURE_SERVICE_URL;
+  if (!baseUrl || !account) {
+    return undefined;
+  }
+
+  try {
+    const data = (
+      await axios.get<{
+        status?: string;
+        description?: string;
+        refers_with_code?: string;
+        wallet_address?: string;
+      }>(`${baseUrl}/get-refers-with/${account}`)
+    )?.data;
+
+    if (!data) {
+      return undefined;
+    }
+    const status = data.status ? parseInt(data.status, 10) : 200;
```

Comment:
I fixed the API to return 404 properly if not found so if you wrote/tested this code before that then the error handling here may need to change.